### ISSUE TITLE
Translation

### ISSUE THREAD

**Issue by abelards** [fibreville/minotaure#100]
Hi all,
it'd be awesome if we could translate all strings.
Do you have a favorite way to do this in PHP?

I mostly use other languages and the awesome "gettext", and I feel like there's not so much text.
I'd be glad to help if you choose and tell me how you want that done.
Cheers,

**Comment by abelards** [comment]
I have skimmed the files and at a first glance, here are the files with french text:
https://github.com/fibreville/minotaure/blob/main/src/variables.php
https://github.com/fibreville/minotaure/blob/main/src/newcomplete.php
https://github.com/fibreville/minotaure/blob/main/src/new.php
https://github.com/fibreville/minotaure/blob/main/src/main.php
https://github.com/fibreville/minotaure/blob/main/src/index.php
https://github.com/fibreville/minotaure/blob/main/src/header.php
https://github.com/fibreville/minotaure/blob/main/src/ecran_forms.php
https://github.com/fibreville/minotaure/blob/main/src/continuevalid.php
https://github.com/fibreville/minotaure/blob/main/src/continue.php
https://github.com/fibreville/minotaure/blob/main/src/connexion_example.php
https://github.com/fibreville/minotaure/blob/main/src/ajax.php
https://github.com/fibreville/minotaure/blob/main/src/admin_tools.php
https://github.com/fibreville/minotaure/blob/main/src/js/main.js

**Comment by uurshin** [comment]
I completed a first crawl of the strings (175 without dupes) and gettext seems to get the job done nicely.
The base language is french because we are very proud.
I've started the FR to EN translation file and pushed it on the dev branch, you can complete the file and attach it to me here, or create a pull request on your own.

Anyone wanting to contribute another translation (french to spanish for example) can ask me to initialize the files.
If you want to do everything by yourself because you're already familiar with how it works, it's ok too !
The only thing you need if you only want to help translating the current string in a language is to install the poedit software and open one of the .po files (for example /src/locale/en/LC_MESSAGES/minotaure.po).

For anyone willing to add future translated strings in the html/php code, three links on the subject  :
- https://www.codeandweb.com/babeledit/tutorials/translation-with-gettext-and-php
- https://phrase.com/blog/posts/learn-gettext-tools-internationalization/#Turning_PO_files_into_MO_files_with_msgfmt
- https://stackoverflow.com/questions/12860888/variables-in-php-gettext

**Comment by abelards** [comment]
J'ai quelques pratiques GetText mais je sais pas si c'est ce que vous souhaitez.
1. je HTML escape tout après donc pas de HTML dans les po
   [cf l22](https://github.com/fibreville/minotaure/blob/dev/src/locale/en/LC_MESSAGES/minotaure.po#L22)
2. je découpe les phrases donc pas de duplicata dans les lignes
   [cf l31 36 41](https://github.com/fibreville/minotaure/blob/dev/src/locale/en/LC_MESSAGES/minotaure.po#31)
3. je ferai des corrections de style plus tard comme ça il y aura plusieurs commits
4. "Votre personnage %s a sombré" ?

**Comment by uurshin** [comment]
Merci pour ce boulot et pour tes remarques pertinentes !

1. Ca rendrait le code plus pénible à lire d'avoir des html entities partout, ça pose quoi comme problème d'avoir du html léger ? (j'ai repéré un souci sur [L502](https://github.com/fibreville/minotaure/blob/dev/src/locale/en/LC_MESSAGES/minotaure.po#319 qu'il faut que je corrige d'ailleurs).
2. Avant le passage sur le mécanisme de trad les phrases étaient découpées en concaténées dynamiquement, je les ai dupliqué pour faciliter la traduction (parce que je pensais que ça donnerait un trad incohérente d'avoir des bouts de phrases). Mais c'était peut être un excès de zèle, je peux redécouper si nécessaire.
3. "Loot" : je crois qu'on a gardé le terme d'origine utilisé par FT. C'est l'attribution d'équipement, de bonus ou autre, donc c'est assez vaste. J'ai pas trop d'idée pour remplacer, hésite pas à proposer.
4. Ok !
5. Oui en passant sur les chaines je me suis fait la réflexion aussi. Il me semble que ce que j'ai fait c'est mettre des points quand on se trouvait dans un endroit un peu "narratif" comme un message qui s'affiche sur la fiche du joueur ou une description , et quand ce sont juste des infos /label de champs je ne mets pas de point. Il y a peu être des exceptions qui font tâche.
6. Carrément ! Pour les exemples cités, je pense qu'il faut retenir "vote" / "votre personnage". Pas trop d'avis sur santé / vie.
7. (et 8) Je suis d'accord, mais effectivement de mon côté j'avais essayé de ne pas effacer toutes les tournures de FT qui donnaient un côté sympathique à la première approche du logiciel.
9. Il faut comprendre "sombré dans la folie" (volonté = 0).
10. Hmm effectivement je comprends pas pourquoi ça a généré ça, je vais regarder (peut être echo qui se comporte différemment de print).
11. Hmm pour harmoniser peut être, mais je trouve strong plus pertinent du coup.

**Comment by abelards** [comment]
1. non mais au lieu d'avoir à traduire "cliquez ici : lien" tu sépares et on traduit "cliquez ici" d'un côté et "lien" de l'autre.
2. OK merci ! L'idée c'est juste que si tu as de "tros gros chunks" ça se voit bien dans le PO que tu traduis XXX + YYY plein de fois

6. bah écoute le point est côté dev principal (= vous en français) et ensuite on verra ;)
7. oui ! Je suis plutôt pour garder ce côté narratif en français et punchy en anglais qui correspondent aux deux cultures

9. j'ai mis failed mais on verra plus tard

11. aucun souci pour "strong" au lieu de "b", j'ai pas d'opinion forte, je remonte juste.

**Comment by uurshin** [comment]
Pour 2 et 6. Yes on fera cette passe dans un second temps.
7. Allez !

**Comment by abelards** [comment]
I scanned the GM screen quickly.
We should probably find a way to set up default values:
- Notre Aventure
- Rejoindre l'Aventure : ... 
- traître
- enregistrer

**Comment by uurshin** [comment]
The GM can edit these strings. Do you mean we should provide translations for the default values. In that case, I agree.

**Comment by uurshin** [comment]
I think that the use of the native php gettext will probably cause issue for some people with no control on their server (each locale need to be installed on the server to work) so I will probably use https://github.com/php-gettext/Translator instead at some point. I will save some headache for many people, but it will add a dependency to the project.
